### ISSUE TITLE
Weird issue when using a password with special character like '['

### ISSUE THREAD

**Issue by alexAubin** [YunoHost/issues#1435]
Can be reproduced by running : 

```
yunohost user create johndoe -f John -l Doe -m johndoe@ynh.dev -p 'weeeeeeird['
```

which displays : `grep: -:1: Invalid regular expression`

because of : https://github.com/YunoHost/yunohost/blob/stretch-testing/src/yunohost/utils/password.py#L178

where grep things we're feeding a regex ... Dunno how to tell grep that it should take it as a raw string :/ 

That doesn't prevent the command from working though...

Alternatively we could ignore this check if there are some special characters since this password list is (I believe) only alphanumeric (+ maybe underscore and #)

**Comment by Psycojoker** [comment]
This is very likely related to the problem of having weird characters in password of applications because we don't correctly handle those situations everywhere we uses bash.

Which has been discussed here https://github.com/YunoHost-Apps/etherpad_mypads_ynh/issues/61 and I suggested a solution https://github.com/YunoHost-Apps/etherpad_mypads_ynh/issues/61#issuecomment-450341737 but there weren't any energy for that.

**Comment by alexAubin** [comment]
This one is not related to apps though, it's really about defining the user password when creating (maybe also updating?) it.

**Comment by Psycojoker** [comment]
Oh ok '-'

**Comment by Saxodwarf** [comment]
There is an easy fix for this one: add the `-F` flag to the grep command, which according to grep's documentation will stop it from interpreting the pattern as a regex:
>```
>       -F, --fixed-strings
>              Interpret PATTERNS as fixed strings, not regular expressions.
>```
The line to change in the source code is, I believe, the following: https://github.com/YunoHost/yunohost/blob/807b577cf27b363bf6db6daade2d6ad836cdd08e/src/yunohost/utils/password.py#L174

**Comment by alexAubin** [comment]
Fixed in 4.1.4
